nikola/nikola.py

```diff
@@ -2374,11 +2376,16 @@ def generic_index_renderer(self, lang, posts, indexes_title, template_name, cont
                 lists.append(posts[:kw["index_display_post_count"]])
                 posts = posts[kw["index_display_post_count"]:]
         num_pages = len(lists)
+        displayed_page_numbers = [utils.get_displayed_page_number(i, num_pages, self) for i in range(max(num_pages, 1))]
+        page_links = [page_link(i, displayed_page_numbers[i], num_pages, False) for i in range(max(num_pages, 1))]
+        if kw['show_index_page_navigation']:
+            temp_map = {page_number - 1: link for page_number, link in zip(displayed_page_numbers, page_links)}
```

Comment:
Why is this a dict? I’d expect the lists to be sorted already, so *this could be a list*. In any case, avoid iterating over `range(num_pages)` more than once — not only you might have missed the case when num_pages = 0, but also `[i[1] for i in sorted(temp_map.values())]` has more relation to the dict.